### PR TITLE
Feature mega menu

### PR DESCRIPTION
**Pull request by MGS-sails**

### CODE REVIEW

File: src/components/Nav/MegaMenu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,693 @@
...
+  const [menuVisible, setMenuVisible] = useState(false);
+  const [activeSection, setActiveSection] = useState<string | null>(null);
+
+  const { dispatch } = useQueryParamsStore();
+
+  const navigate = useNavigate();
+
+  const handleMouseEnter = (sectionId: string) => {
+    setMenuVisible(true);
+    setActiveSection(sectionId);
```
Comment (SandyRogers) [inline src/components/Nav/MegaMenu/index.tsx:21]: I think we need something here to only set the active section if there is content for it. Currently, if you hover over e.g. `Submit data`, and then move mouse sideways to `Overview`, the `Submit data` megamenu section still shows.

**Review by SandyRogers — COMMENTED** [review]
This looks great, thank you @MGS-sails 

I have tried it out locally. I left one comment inline about the behaviour navigating from an expandable section to a non expandable section – not a huge deal, but if there is a simple fix I think that is worth doing.

Only other very minor comment is that perhaps the external link icons are a bit excessive within the mega menu? E.g. on the `Help` section where ~everything is an external link, they feel a bit disruptive. 

See what you think; these are totally subjective!

File: src/components/Nav/MegaMenu/index.tsx (hunk 1 above)
Comment (MGS-sails) [reply src/components/Nav/MegaMenu/index.tsx:21]: Thanks @SandyRogers . I just pushed a fix to hide the megamenu content, for links that have no content section.

With regards to the external link icons. I can see what you're saying, however I think it's important to indicate when links open in a new tab. I just used that Icon cuz I saw it was already being used elsewhere in the app. 
But perhaps in the future we can choose a more subtle icon.

**Review by SandyRogers — APPROVED** [review]
Nice, thanks!

**Pull request merged**